packaging/nix/flake.nix

```
@@ -2,55 +2,50 @@
   description = "umu universal game launcher";
 
   inputs = {
-    nixpkgs = {
-      type = "github";
-      owner = "NixOS";
-      repo = "nixpkgs";
-      ref = "nixpkgs-unstable";
-    };
+    nixpkgs.url = "github:NixOS/nixpkgs/nixpkgs-unstable";
   };
 
   outputs = {
     self,
     nixpkgs,
   }: let
-    umu-launcher-src = builtins.toPath "${self}/../../";
-
-    pkgs = import nixpkgs {
-      system = "x86_64-linux";
-      overlays = [self.overlays.default];
-    };
-
-    version = "1.1.4";
+    inherit (nixpkgs) lib;
+
+    # Utility function for producing consistent rename warning messages
+    rename = old: new: lib.warn "`${old}` has been renamed to `${new}`";
+
+    # Package sets supported by umu-launcher-unwrapped
+    # { system = pkgs; }
+    supportedPkgs =
+      lib.filterAttrs
+      (system: pkgs: builtins.elem system pkgs.umu-launcher-unwrapped.meta.platforms)
+      nixpkgs.legacyPackages;
+
+    # Generate a version string, based on the flake's lastModifiedDate
+    version = let
+      date = self.lastModifiedDate;
+      parts = builtins.match "(\[0-9\]{4})(\[0-9\]{2})(\[0-9\]{2})(\[0-9\]{2})(\[0-9\]{2})(\[0-9\]{2})" date;
+    in "git.${lib.concatStringsSep "." parts}";
   in {
-    overlays.default = final: prev: let
-      py = prev.python3;
-    in {
-      umu-launcher = final.callPackage ./umu-launcher.nix {
+    overlays.default = final: prev: {
+      umu-launcher = final.callPackage ./package.nix {
         inherit version;
-        umu-launcher = umu-launcher-src;
-        pyth1 = py;
-      };
-
-      umu-run = final.callPackage ./umu-run.nix {
-        inherit version;
-        package = final.umu-launcher;
-      };
-
-      umu = final.callPackage ./combine.nix {
-        inherit version;
-        env = final.umu-run;
-        package = final.umu-launcher;
-        truststore = true;
-        cbor2 = true;
+        inherit (prev) umu-launcher-unwrapped;
       };
+      umu = rename "umu" "umu-launcher" final.umu-launcher;
+      umu-run = rename "umu-run" "umu-launcher" final.umu-launcher;
+      inherit (final.umu-launcher.passthru) umu-launcher-unwrapped;
     };
 
     formatter = builtins.mapAttrs (system: pkgs: pkgs.alejandra) nixpkgs.legacyPackages;
 
-    packages.x86_64-linux = {
-      inherit (pkgs) umu;
-      default = self.packages.x86_64-linux.umu;
-    };
+    packages =
+      builtins.mapAttrs (system: pkgs: rec {
+        default = umu-launcher;
+        umu-launcher = pkgs.callPackage ./package.nix {inherit version;};
+        umu = rename "packages.${system}.umu" "packages.${system}.umu-launcher" umu-launcher;
+        inherit (umu-launcher.passthru) umu-launcher-unwrapped;
```

Can we not avoid this passthru thing?

---

I usually prefer to be explicit, but yes; passthru attrs are also present directly on the drv-attrs.

So this is effectively the same thing:

```suggestion
        inherit (umu-launcher) umu-launcher-unwrapped;
```

---

I meant get rid of the passthru in umu-launcher enitrely, and get the unwrapped paxkage directly.

---

The (overridden) unwrapped package is defined in `./package.nix`, so we need passthru in order to access it.

If you used `pkgs.umu-launcher-unwrapped`, you'd get the non-overridden package.

---

Oh? Why isn't there a separate umu-launcher-wrapped package exposed?

---

There is. It is _overridden_ in `./package.nix`, and then _exposed_ to the overlay and the packages output, _via_ `umu-launcher.passthru`.

It would also be possible to have a separate file where the unwrapped package is overridden, however this felt messier because most of the overrides are based on arguments to the `umu-launcher` package.

---

There is a reason umu-unwrapped is separate in nixpkgs. I believe there is a use case, where users may want to use the unwrapped package over the wrapper one.

---

That is precisely why it _is_ separately exposed in the overlay and flake outputs.

---

Oh, my mistake. I still think it makes sense to separate it to a different file though.

---

I'll play around with it locally and push something up _if_ it ends up being an overall improvement.

---

I've pushed this as a `fixup` commit.

If reviewers prefer it to the prior impl I'll squash.

---

I like it